File: core/src/com/symbol/game/ecs/entity/MapEntityType.java
package com.symbol.game.ecs.entity;

public enum MapEntityType {

    None("none"),
    MovingPlatform("mplatform"),
    TemporaryPlatform("tplatform"),
    Portal("portal"),
    Clamp("clamp"),
    HealthPack("hp_pack"),
    Mirror("mirror"),
    GravitySwitch("gswitch"),
    SquareSwitch("sswitch"),
    ToggleTile("toggle");

    public String typeStr;

    MapEntityType(String typeStr) {
        this.typeStr = typeStr;
    }

    public static MapEntityType getType(String typeStr) {
        for (MapEntityType type : MapEntityType.values()) {
            if (type.typeStr.equals(typeStr)) return type;
        }
        return null;
    }

}

File: core/src/com/symbol/game/ecs/entity/MapEntityType.kt
package com.symbol.game.ecs.entity

enum class MapEntityType(val typeStr: String) {

    None("none"),
    MovingPlatform("mplatform"),
    TemporaryPlatform("tplatform"),
    Portal("portal"),
    Clamp("clamp"),
    HealthPack("hp_pack"),
    Mirror("mirror"),
    GravitySwitch("gswitch"),
    SquareSwitch("sswitch"),
    ToggleTile("toggle"),
    ForceField("force_field"),
    DamageBoost("damage_boost"),
    InvertSwitch("iswitch");

    companion object {
        fun getType(typeStr: String) : MapEntityType? = values().find { it.typeStr == typeStr }
    }

}